rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -164,28 +194,25 @@ def _recursively_initialize_decoders(
                         )
                         continue
 
-                    address_results.update(self.decoders[class_name].addresses_to_decoders())
-                    rules_results.extend(self.decoders[class_name].decoding_rules())
-                    enricher_results.extend(self.decoders[class_name].enricher_rules())
-                    self.all_counterparties.update(self.decoders[class_name].counterparties())
+                    results.address_mappings.update(self.decoders[class_name].addresses_to_decoders())  # noqa: E501
+                    results.event_rules.extend(self.decoders[class_name].decoding_rules())
+                    results.token_enricher_rules.extend(self.decoders[class_name].enricher_rules())
+                    results.post_decoding_rules.extend(self.decoders[class_name].post_decoding_rules())  # noqa: E501
+                    results.counterparties.update(self.decoders[class_name].counterparties())
 
-                recursive_addrs, recursive_rules, recurisve_enricher_results = self._recursively_initialize_decoders(full_name)  # noqa: E501
-                address_results.update(recursive_addrs)
-                rules_results.extend(recursive_rules)
-                enricher_results.extend(recurisve_enricher_results)
+                recursive_results = self._recursively_initialize_decoders(full_name)
+                results += recursive_results
 
-        return address_results, rules_results, enricher_results
+        results.post_decoding_rules.sort(key=lambda x: x[1], reverse=True)
+        return results
 
     def initialize_all_decoders(self) -> None:
         """Recursively check all submodules to get all decoder address mappings and rules
         """
         self.decoders: dict[str, 'DecoderInterface'] = {}
-        address_result, rules_result, enrichers_result = self._recursively_initialize_decoders(self.chain_modules_root)  # noqa: E501
-        self.address_mappings = address_result
-        self.event_rules.extend(rules_result)
-        self.token_enricher_rules.extend(enrichers_result)
-        # update with counterparties not in any module
-        self.all_counterparties.update(self.misc_counterparties)
+        rules = self._recursively_initialize_decoders(self.chain_modules_root)
```

Comment:
I think this function is no longer needed. Include its contents directly in the constructor, the only place it's called.

Reply:
Makes sense :+1: